### PR TITLE
refactor: add permission checkup in onGetContacts in UserInfoBridgeDispatcher

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
* In `UserInfoBridgeDispatcher.onGetContacts` PR, we haven't added the permission checkup for `MiniAppCustomPermissionType.CONTACT_LIST` like the user-name and profile photo we have. This PR aims to add the fix. 
* Add a common function e.g. `hasPermission` in `MiniAppCustomPermissionCache` for checking if any specific permission has been allowed or not.

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by corycaywood — APPROVED** [review]
Looks good 👍 

You could also use the new `hasPermission` in `CustomPermissionBridgeDispatcher`: 
https://github.com/rakutentech/android-miniapp/blob/master/miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/CustomPermissionBridgeDispatcher.kt#L70

Also, you can apply this to the geo location request after that PR is merged.

**Comment by khairul-alam-licon** [comment]
> You could also use the new hasPermission in CustomPermissionBridgeDispatcher


Thanks for this suggestion. I have used new `hasPermission` in CustomPermissionBridgeDispatcher:
https://github.com/rakutentech/android-miniapp/blob/master/miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/CustomPermissionBridgeDispatcher.kt#L70

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/userinfo/UserInfoBridgeDispatcher.kt [hunk 1]
```diff
@@ -65,11 +64,10 @@ abstract class UserInfoBridgeDispatcher {
...
-            customPermissionCache.readPermissions(miniAppId).pairValues.find {
-                it.first == MiniAppCustomPermissionType.USER_NAME &&
-                        it.second == MiniAppCustomPermissionResult.ALLOWED
-            }?.let { isPermissionGranted = true }
+            val isPermissionGranted = customPermissionCache.hasPermission(
+                miniAppId,
+                MiniAppCustomPermissionType.USER_NAME
+            )
 
             if (isPermissionGranted) {
```
Comment (minh-rakuten) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/userinfo/UserInfoBridgeDispatcher.kt:72]: you can call `customPermissionCache.hasPermission` here, just like what you did in `MiniAppChromeClient`.

**Review by minh-rakuten — APPROVED** [review]
👍

**Pull request merged**